it/common/src/main/java/org/apache/beam/it/common/matchers/LaunchInfoSubject.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.it.common.matchers;
+
+import com.google.common.truth.FailureMetadata;
+import com.google.common.truth.Subject;
+import org.apache.beam.it.common.PipelineLauncher.JobState;
+import org.apache.beam.it.common.PipelineLauncher.LaunchInfo;
+
+/**
+ * Subject that has assertion operations for {@link LaunchInfo}, which has the information for a
+ * recently launched pipeline.
+ */
+public final class LaunchInfoSubject extends Subject {
+
+  private final LaunchInfo actual;
+
+  private LaunchInfoSubject(FailureMetadata metadata, LaunchInfo actual) {
+    super(metadata, actual);
```

Comment:
Using `super` constructor just to initialize fields is an anti-pattern. It seems to be built into Truth, which is a bummer. Is this the intended use of `Subject`? Is there a way to use composition instead of inheritance to more cleanly separate this class?

Reply:
Yeah, it is built into Truth. Inheritance is the recommended way of using `Subject`. See https://truth.dev/extension.html